### PR TITLE
Send Email Before Application Expired

### PR DESCRIPTION
**Pull request by knockknockyoo**

### CODE REVIEW

File: services/application.js [hunk 1]
```diff
@@ -303,27 +303,28 @@ class Application {
         }
     }
 
+    async remindApplicationSubmission() {
+        const inactiveDuration = this.emailParams.inactiveDays - this.emailParams.remindDay;
```
Comment (AustinSMueller) [inline services/application.js:307]: This logic is incorrect. If use default values, inactiveDays is 45 and remnidDay is 30 so inactiveDuration is 15 and we send a reminder after 15 days. We want to send the reminder after 30 days so we can remove this "inactiveDuration" calculation and just use the remindDay value instead

File: services/application.js [hunk 2]
```diff
@@ -303,27 +303,28 @@ class Application {
...
+        const remindCondition = {
+            updatedAt: {
+                $lt: subtractDaysFromNow(inactiveDuration),
```
Comment (AustinSMueller) [inline services/application.js:310]: This was not working for me, the query was always returning 0 applications. I think the issue is that subtractDaysFromNow is returning a date object. I was able to get it to work by converting to an ISO string before using it in the pipeline here.  ".toISOString()"

**Review by AustinSMueller — CHANGES_REQUESTED** [review]

File: app.js [hunk 3]
```diff
@@ -55,12 +55,14 @@ cronJob.schedule(config.schedule_job, async () => {
...
         const organizationCollection = new MongoDBCollection(dbConnector.client, DATABASE_NAME, ORGANIZATION_COLLECTION);
-        const emailParams = {url: config.emails_url, officialEmail: config.official_email, inactiveDays: config.inactive_user_days};
+        const emailParams = {url: config.emails_url, officialEmail: config.official_email, inactiveDays: config.inactive_user_days, remindDay: config.remind_application_days};
         const logCollection = new MongoDBCollection(dbConnector.client, DATABASE_NAME, LOG_COLLECTION);
         const userService = new User(userCollection, logCollection);
         const dataInterface = new Application(logCollection, applicationCollection, new Organization(organizationCollection), userService, dbService, notificationsService, emailParams);
         console.log("Running a scheduled background task to delete inactive application at " + getCurrentTime());
         await dataInterface.deleteInactiveApplications(config.inactive_user_days);
+        console.log("Running a scheduled background task to remind inactive application at " + getCurrentTimeYYYYMMDDSS());
```
Comment (AustinSMueller) [inline app.js:64]: This  getCurrentTimeYYYYMMDDSS() needs to be updated to getCurrentTime()

File: services/application.js (hunk 2 above)
Comment (knockknockyoo) [reply services/application.js:310]: updated it to 30days.

File: services/application.js (hunk 1 above)
Comment (knockknockyoo) [reply services/application.js]: updated it to 30 days.

**Review by AustinSMueller — APPROVED** [review]

**Pull request merged**